lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+gem 'pg'
+require 'pg'
+
+module Superstore
+  module Adapters
+    class JsonbAdapter < AbstractAdapter
+      class QueryBuilder
+        def initialize(adapter, scope)
+          @adapter  = adapter
+          @scope    = scope
+        end
+
+        def to_query
+          [
+            "SELECT #{select_string} FROM #{@scope.klass.table_name}",
+            where_string,
+            order_string,
+            limit_string
+          ].delete_if(&:blank?) * ' '
+        end
+
+        def select_string
+          if @scope.select_values.any?
+            "id, jsonb_slice(attribute_store, #{@adapter.fields_to_postgres_array(@scope.select_values)}) as attribute_store"
+          else
+            '*'
+          end
+        end
+
+        def where_string
+          wheres = @scope.where_values.dup
+          if @scope.id_values.any?
+            wheres << @adapter.create_ids_where_clause(@scope.id_values)
+          end
+
+          if wheres.any?
+            "WHERE #{wheres * ' AND '}"
+          end
+        end
+
+        def order_string
+          if @scope.order_values.any?
+            orders = @scope.order_values.join(', ')
+            "ORDER BY #{orders}"
+          elsif @scope.id_values.many?
+            id_orders = @scope.id_values.map { |id| "ID=#{@adapter.quote(id)} DESC" }.join(',')
+            "ORDER BY #{id_orders}"
+          end
+        end
+
+        def limit_string
+          if @scope.limit_value
+            "LIMIT #{@scope.limit_value}"
+          end
+        end
+      end
+
+      def primary_key_column
+        'id'
+      end
+
+      def connection
+        # conf = {:adapter=>"postgresql", :encoding=>"unicode", :database=>"axle_place_test", :pool=>5, :username=>"postgres"}
+        # @connection ||= ActiveRecord::Base.postgresql_connection(conf)
+        ActiveRecord::Base.connection
+      end
+
+      def execute(statement)
+        ActiveSupport::Notifications.instrument("cql.cassandra_object", cql: statement) do
```

Comment:
Ah good point. That would require a change in all the other adapters too. Our code listens to these types of queries for spokes https://github.com/data-axle/pipeline/blob/master/lib/place_directory/collect_metrics.rb#L80-L90